src/components/TodoInfo/TodoInfo.tsx

```diff
@@ -0,0 +1,141 @@
+import cn from 'classnames';
+import {
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo
+  onUpdate: (todo: Todo, prop: string, value: any) => void
+  onDelete: (todoId: number) => void
+  changingTodoIds: number | number[]
+};
+
+export const TodoInfo: React.FC<Props> = ({
+  todo,
+  onUpdate,
+  onDelete,
+  changingTodoIds,
+}) => {
+  const [newTitle, setNewTitle] = useState(todo.title);
+  const [isEditing, setIsEditing] = useState(false);
+  const newTodoField = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (newTodoField.current) {
+      newTodoField.current.focus();
+    }
+  }, [isEditing]);
+
+  const titleEditHandler = () => {
+    if (!newTitle.length) {
+      onDelete(todo.id);
+    } else if (newTitle !== todo.title) {
+      onUpdate(todo, 'title', newTitle);
+    }
+
+    setIsEditing(false);
+  };
+
+  return (
+    <div
+      data-cy="Todo"
+      className={cn('todo', {
+        completed: todo.completed,
+      })}
+      key={todo.id}
```

Comment:
you're dropping too much `key` attributes
you don't seem to even need any here
you only need to provide it when `.map` in JSX
also, you're using the same (hence, NOT UNIQUE) `key` few times - which is wrong
key attribute must be unique

reconsider it's usage throughout your code